xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -0,0 +1,316 @@
+		let num_events2 = frame_system::Pallet::<T>::events().len();
+		assert_eq!(num_events + 1, num_events2);
+	}
+
+	refund_surplus {
+		let mut executor = new_executor::<T>(Default::default());
+		executor.total_surplus = 1337;
+		executor.total_refunded = 0;
+
+		let instruction = Instruction::<XcmCallOf<T>>::RefundSurplus;
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		let result = executor.execute(xcm)?;
+	} verify {
+		assert_eq!(executor.total_surplus, 1337);
+		assert_eq!(executor.total_refunded, 1337);
+	}
+
+	set_error_handler {
+		let mut executor = new_executor::<T>(Default::default());
+		let instruction = Instruction::<XcmCallOf<T>>::SetErrorHandler(Xcm(vec![]));
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert_eq!(executor.error_handler, Xcm(vec![]));
+	}
+
+	set_appendix {
+		let mut executor = new_executor::<T>(Default::default());
+		let appendix = Xcm(vec![]);
+		let instruction = Instruction::<XcmCallOf<T>>::SetAppendix(appendix);
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert_eq!(executor.appendix, Xcm(vec![]));
+	}
+
+	clear_error {
+		let mut executor = new_executor::<T>(Default::default());
+		executor.error = Some((5u32, XcmError::Overflow));
+		let instruction = Instruction::<XcmCallOf<T>>::ClearError;
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert!(executor.error.is_none())
+	}
+
+	descend_origin {
+		let mut executor = new_executor::<T>(Default::default());
+		let who = X2(OnlyChild, OnlyChild);
+		let instruction = Instruction::DescendOrigin(who.clone());
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert_eq!(
+			executor.origin,
+			Some(MultiLocation {
+				parents: 0,
+				interior: who,
+			}),
+		);
+	}
+
+	clear_origin {
+		let mut executor = new_executor::<T>(Default::default());
+		let instruction = Instruction::ClearOrigin;
+		let xcm = Xcm(vec![instruction]);
+	} : {
+		executor.execute(xcm)?;
+	} verify {
+		assert_eq!(executor.origin, None);
+	}
+
+	report_error {
+		let mut executor = new_executor::<T>(Default::default());
+		executor.error = Some((0u32, XcmError::Unimplemented));
+		let query_id = Default::default();
+		let dest = T::valid_destination().map_err(|_| BenchmarkError::Skip)?;
+		let max_response_weight = Default::default();
+
+		let instruction = Instruction::ReportError { query_id, dest, max_response_weight };
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// the execution succeeding is all we need to verify this xcm was successful
+	}
+
+	claim_asset {
+		let (origin, ticket, assets) = T::claimable_asset()
+			.ok_or(BenchmarkError::Skip)?;
+
+		// We place some items into the asset trap to claim.
+		let mut executor = new_executor::<T>(origin.clone());
+		executor.holding = assets.clone().into();
+		executor.execute(Xcm(vec![]))?;
+		executor.post_execute(0);
```

Comment:
```suggestion
		<T as Config>::XcmConfig::AssetTrap::drop_assets(origin, assets);
```
mmh, if you're going below the abstractions why not call `<T as Config>::XcmConfig::AssetTrap::drop_assets(origin, assets)` directly?

Reply:
done!